Add dashboard screen

maxpluez opened the pull request:
Note: I need to make "scanned" a global state, because multiple screens and components need to update that to false so the QR scanner will scan again when we go back to that screen.

dvorjackz commented:
> Note: I need to make "scanned" a global state, because multiple screens and components need to update that to false so the QR scanner will scan again when we go back to that screen.

https://www.notion.so/uclabruinbot/d22558f5479944f1a7f74a025dcdd703?v=684f4fba77a64fa5aca5a4edb21f764e&p=0b407ce4c7fa468a87e0d043fbf79989

dvorjackz commented:
> Note: I need to make "scanned" a global state, because multiple screens and components need to update that to false so the QR scanner will scan again when we go back to that screen.

For features let's have a screenshot in the description?

And let's get some information in the PR's too, helps with reviewing. e.g.

## Changes
- Changed this
- Changed that

## Screenshots
<some-image.jpg>

maxpluez commented:
Changes:
1. Added a dashboard screen, updated App.tsx to include this screen in the stack.
2. Make scanned a global state so that when the other screen navigates back to QrScreen, they can set scanned to false to re-scan.
3. Fix deeplink.

maxpluez commented:
![35F0F3A3-9F6B-45FC-BDA3-B63022EDBB48](https://user-images.githubusercontent.com/25847912/102003786-7c7e3c80-3cbf-11eb-8fd6-0e11d0f9ccb8.png)

zhang-edward commented on src/components/QrView.tsx:
```diff
@@ -19,14 +19,12 @@ interface Props {
 	const [hasCameraPermission, setCameraPermission] = useState('null');
-	const [scanned, setScanned] = useState(false);
 
 	const { state, dispatch } = useContext(Ctx);
```
Why do we use application-level state for this?

maxpluez replied:
Because many screens will need the ability to go back to "scan QR code". I tested myself that if we go (navigate) back to the QR screen from another screen, the QR screen isn't reloaded, so the local component level "scanned" state is still true, and the scanner will not scan if "scanned" is true. Putting setScannned(false) in useEffect won't work since the QR component is not reloaded, so it is the job of the other screen to update scanned.

dvorjackz replied:
Wait ignore my approval idk how to remove it. Actually, I don't think it would be good to get other screens to set scanned back to false. I'm sure there's probably a way to force run the react hook constructor again. Or if you really want to do it this way, would you not need to put it in context instead?

dvorjackz reviewed (dismissed):
LGTM!

maxpluez commented:
Accidentally clicked on resolve conversation.. @dvorjackz @zhang-edward I found another solution and removed the global scanned state. Have a look!

zhang-edward commented on src/components/QrView.tsx:
```diff
@@ -51,7 +50,7 @@ const QrComponent = ({ navigateForward }: Props) => {
 			<BarCodeScanner
 				onBarCodeScanned={
-					scanned
+					scanned || state.bot
```
Actually, why do we need `scanned` in the first place? If we don't have this flag, will the barcode scanner just scan continuously?

dvorjackz commented on App.tsx:
```diff
@@ -41,31 +45,33 @@ const Home = () => {
+				});
 		},
-		[dispatch, state.bot]
+		[dispatch]
```
Jw, how does this work again? Shouldn't this be like botId or something?
